camel/o1datagen/o1datagen.py

```diff
@@ -0,0 +1,332 @@
+            raise ValueError(
+                f"""Golden answer for question '{question}' is empty.
+                Cannot proceed with verification."""
+            )
+        prompt = f"""Please determine if the following two answers 
+        express the same meaning:
+        Question: {question}
+        Answer 1: {answer}
+        Answer 2: {golden_answer}
+        Just answer "True" or "False".
+        """
+        response = self.chat_agent.step(prompt)
+        is_correct = response.msgs[0].content.strip().lower() == "true"
+        logger.info("Answer verification result: %s", is_correct)
+        return is_correct
+
+    def monte_carlo_tree_search(
+        self, question: str, partial_solution: str = ""
+    ) -> tuple[str, bool]:
+        r"""
+        Generate and verify answers using Monte Carlo Tree Search.
+
+        This method implements a Monte Carlo Tree Search approach
+        to find the best solution
+        by iteratively generating answers and scoring them against
+        the golden answer.
+
+        Args:
+            question (str): The problem or question to be solved.
+            partial_solution (str, optional): A partial solution to build upon.
+             This canbe used to guide the search process
+             with existing progress.
+             Defaults to an empty string.
+
+        Returns:
+            tuple[str, bool]: A tuple containing:
+                - str: The best solution found
+                (or empty string if no solution found)
+                - bool: True if a correct solution was found, False otherwise
+        """
+        logger.info("Starting Monte Carlo Tree Search")
+        best_solution = ""  # Initialize as empty string instead of None
+        best_score: float = 0.0
+        for i in range(self.search_limit):
+            # Generate new answer
+            current_solution = self.get_answer(question, partial_solution)
+            # Verify answer
+            is_correct = self.verify_answer(question, current_solution)
+            if is_correct:
+                logger.info("Correct answer found! Stopping search")
+                return current_solution, True
+            prompt = f"""Analyze  the similarity of this answer 
+            to the correct answer (between 0-1):
+            Question: {question}
+            Generated answer: {current_solution}
+            Correct answer: {self.golden_answers[question]}
+            Just return a number between 0-1.
+            """
+            response = self.chat_agent.step(prompt)
+            try:
+                score = float(response.msgs[0].content.strip())
+                if score > best_score:
+                    best_score = score
+                    best_solution = current_solution
+                logger.info(
+                    "Current search progress: %d/%d, best score: %.2f",
+                    i + 1,
+                    self.search_limit,
+                    best_score,
+                )
```

Comment:
if `current_solution` has a very high score(> best_score), should we exit the loop?

Reply:
sounds good ，done it